build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.build-logic.groovy-dsl-gradle-plugin.gradle.kts

```diff
@@ -28,7 +28,7 @@ java {
 
 dependencies {
     api(platform(project(":build-platform")))
-    implementation("gradlebuild:code-quality")
+    implementation("gradlebuild:gradle-plugin")
```

Comment:
Do you know why gradle plugins in `build-logic` depend on the `gradle-plugin` project in `build-logic-commons`? I can see how the build script depends on it, but why does the build logic in `build-logic` depend on `gradle-plugin`?

Reply:
For example, the build logic in `build-logic` depends on `TestFilesCleanupPlugin`?

Reply:
Right.

Reply:
I wonder if it would make sense to only add the dependency in the `build-logic` subproject which needs it, and not to all the subprojects.

Reply:
Which one doesn't need it? We seem to apply `ci-reporting` plugin to all `build-logic` subprojects, doesn't everyone need it?